crates/nu-command/src/conversions/into/record.rs

```diff
@@ -103,45 +100,61 @@ impl Command for SubCommand {
     }
 }
 
-fn into_record(
-    engine_state: &EngineState,
-    call: &Call,
-    input: PipelineData,
-) -> Result<PipelineData, ShellError> {
-    let input = input.into_value(call.head)?;
-    let input_type = input.get_type();
-    let span = input.span();
-    let res = match input {
-        Value::Date { val, .. } => parse_date_into_record(val, span),
-        Value::Duration { val, .. } => parse_duration_into_record(val, span),
-        Value::List { mut vals, .. } => match input_type {
-            Type::Table(..) if vals.len() == 1 => vals.pop().expect("already checked 1 item"),
-            _ => Value::record(
-                vals.into_iter()
-                    .enumerate()
-                    .map(|(idx, val)| (format!("{idx}"), val))
-                    .collect(),
-                span,
-            ),
-        },
-        Value::Range { val, .. } => Value::record(
-            val.into_range_iter(span, engine_state.signals().clone())
-                .enumerate()
-                .map(|(idx, val)| (format!("{idx}"), val))
-                .collect(),
+fn into_record(call: &Call, input: PipelineData) -> Result<PipelineData, ShellError> {
+    let span = input.span().unwrap_or(call.head);
+    match input {
+        PipelineData::Value(Value::Date { val, .. }, _) => {
+            Ok(parse_date_into_record(val, span).into_pipeline_data())
+        }
+        PipelineData::Value(Value::Duration { val, .. }, _) => {
+            Ok(parse_duration_into_record(val, span).into_pipeline_data())
+        }
+        PipelineData::Value(Value::List { .. }, _) | PipelineData::ListStream(..) => {
+            let mut record = Record::new();
+            let metadata = input.metadata();
+            for item in input.into_iter() {
+                // Don't use .extend() unless that gets changed to check for duplicate keys
+                let span = item.span();
+                match item {
+                    Value::Record { val, .. } => {
+                        for (key, val) in val.into_owned() {
+                            record.insert(key, val);
+                        }
```

Comment:
Should the two new forms you are propsosing be mutually exclusive? (so either `[[a b] [c d]]` or [{a: b} {c: d}]` instead of also permitting `[[a b] {c: d}]`)

In principle we could create a useable result even with a mixed list, but maybe the mixed form is more indicative of a potential mistake?

Reply:
Yeah, I could lock in one or the other based on the first element seen. I think it might be useful for catching problems